### PR TITLE
Adds new accidents to the omen component and makes it an admin smite

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

The omen component now has a 15% chance when walking through an airlock to have it try to crush you and a 15% chance when walking next to an open turf to throw yourself into it alongside the existing 15% chance to get crushed by a vending machine and 50% chance when you trip over to crack your skull open.

It is also an admin smite with an option to make it permanent.

Allows airlocks to override safties for a single close() proc call, forcing a crush.

## Why It's Good For The Game

More admin smites good. A really good meme smite to throw on someone and watch them suffer accidents for the rest of the round or to put on them for a one-off accident.

## Changelog
:cl:
tweak: Even more dark omens can befall anyone unlucky enough to get a cursed coupon...
add: Admins can now smite players with Bad Luck to make them rather accident prone.
/:cl:

### CODE REVIEW

**Comment by ArcaneMusic** [comment]
Admins, right now:
image

File: code/datums/components/omen.dm [hunk 1]
```diff
@@ -43,14 +53,41 @@
...
 	if(!prob(15))
 		return
-	for(var/t in get_adjacent_open_turfs(our_guy))
+	var/our_guy_pos = get_turf(living_guy)
+	for(var/turf_content in our_guy_pos)
+		if(istype(turf_content, /obj/machinery/door/airlock))
+			to_chat(living_guy, "<span class='warning'>A malevolent force launches your body to the floor...</span>")
+			var/obj/machinery/door/airlock/darth_airlock = turf_content
+			living_guy.apply_status_effect(STATUS_EFFECT_KNOCKDOWN, 10, TRUE)
+			living_guy.apply_status_effect(STATUS_EFFECT_STUN, 10, TRUE)
```
Comment (Rohesie) [inline code/datums/components/omen.dm:69]: Why don't you add a "wormed" new status effect for this, with both the effects of knockdown and stun?

Comment (Timberpoes) [reply code/datums/components/omen.dm]: Wormed status effect is now implemented!

File: code/__DEFINES/traits.dm [hunk 2]
```diff
@@ -87,6 +87,8 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_PULL_BLOCKED		"pullblocked"
 /// Abstract condition that prevents movement if being pulled and might be resisted against. Handcuffs and straight jackets, basically.
 #define TRAIT_RESTRAINED		"restrained"
+/// A mix of stunned and floored. Immobilises and incapacitates the target in a prone position on the floor.
+#define TRAIT_WORMED			"wormed"
```
Comment (Rohesie) [inline code/__DEFINES/traits.dm:91]:
```suggestion
```
No need for this, it's a status effect, not a trait.

File: code/_globalvars/traits.dm [hunk 3]
```diff
@@ -10,6 +10,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_FLOORED" = TRAIT_FLOORED,
 		"TRAIT_HANDS_BLOCKED" = TRAIT_HANDS_BLOCKED,
 		"TRAIT_RESTRAINED" = TRAIT_RESTRAINED,
+		"TRAIT_WORMED" = TRAIT_WORMED,
```
Comment (Rohesie) [inline code/_globalvars/traits.dm:13]:
```suggestion
```

File: code/datums/status_effects/debuffs.dm [hunk 4]
```diff
@@ -958,3 +957,24 @@
...
+	id = "wormed"
+
+/datum/status_effect/incapacitating/wormed/on_apply()
+	. = ..()
+	if(!.)
+		return
+	ADD_TRAIT(owner, TRAIT_FLOORED, TRAIT_STATUS_EFFECT(id))
+	ADD_TRAIT(owner, TRAIT_INCAPACITATED, TRAIT_STATUS_EFFECT(id))
+	ADD_TRAIT(owner, TRAIT_IMMOBILIZED, TRAIT_STATUS_EFFECT(id))
+	ADD_TRAIT(owner, TRAIT_HANDS_BLOCKED, TRAIT_STATUS_EFFECT(id))
```
Comment (Rohesie) [inline code/datums/status_effects/debuffs.dm:972]: Wait, I thought you wanted to drop them to the floor and impede their hand usage. I didn't notice you wanted to also immobilize them, else you just have the paralyzed condition, don't you?

**Comment by Timberpoes** [comment]
Updated to use paralysed status effect instead.

**Review by Rohesie — APPROVED** [review]

**Pull request merged**